### ISSUE TITLE
Auto-scaling font sizes

### ISSUE THREAD

**Issue by dangoldin** [gnab/remark#159]
I've been using Remark for a class I've been teaching after wanting to take advantage of syntax highlighting and think it's awesome. One thing I wish it had was the ability to automatically resize font sizes on each slide similar to what Keynote does.

I'll be glad to take a stab at this if people think it's a decent idea and it would definitely be implemented as a separate option.

**Comment by dangoldin** [comment]
I'm definitely new to the JS world though so would need some review and oversight. Will take a stab at it.

**Comment by kgaut** [comment]
subscribing !

**Comment by junderhill** [comment]
Did you make any progress with this feature? If not I may take a look.
Perhaps we could utilise a library such as https://github.com/davatron5000/FitText.js in order to fulfil this feature?

Another such library is http://www.zachleat.com/web/bigtext-makes-text-big/

**Comment by dangoldin** [comment]
Nope - I wasn't able to get much done unfortunately.

**Comment by junderhill** [comment]
OK, I'm going to take a look at this and see what I can do.

**Comment by junderhill** [comment]
@gnab What's your thoughts on utilising jQuery to support this? It seems that all the libraries that would offer this functionality depend on jQuery.

**Comment by gnab** [comment]
It would be nice if jQuery wasn't required to use remark. If jQuery is needed for this feature, maybe it can be an optional plugin that depends on jQuery, that is not bundled by default. If you can make it work the way you want with jQuery first, we'll take it from there. Maybe we can bundle only the parts of jQuery that is needed, or even implement it ourselves.

**Comment by junderhill** [comment]
I can understand that. I will see what aspects of jQuery it utilises and see if it's something we can reimplement using standard JavaScript.

**Comment by gnab** [comment]
The `updateDimensions` is called to set the dimensions of a single slide:

https://github.com/gnab/remark/blob/develop/src/remark/views/slideView.js#L29

The dimensions remain the same unless the ratio of the slideshow is modified after creation. Window resizing is handled by scaling the entire slide, so font-sizing should only be needed in after the slide's dimensions are set/updated in the above function.

**Comment by aseigneurin** [comment]
Does anyone have a solution for this issue? Maybe a prototype of implementation?

**Comment by kgaut** [comment]
Hi @aseigneurin 

Personnaly I moved to reveal.js for my presentations.

**Comment by aseigneurin** [comment]
@kgaut: do you imply that reveal.js does auto-scaling?

**Comment by Vebryn** [comment]
According reveal.js README, it does: https://github.com/hakimel/reveal.js/#presentation-size

**Comment by koppor** [comment]
Quick solution using [jquery-quickfit](https://github.com/chunksnbits/jquery-quickfit):

```javascript
    <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.7.2/jquery.min.js" ></script>
    <script src="https://cdn.rawgit.com/chunksnbits/jquery-quickfit/master/jquery.quickfit.js"></script>
    <script>
      var slideshow = remark.create();
      $('.remark-slide-content').quickfit({'min': 10, 'max': 36});
    </script>
```

In case the text still flows out, just replace `10` by a smaller minimum size.

**Comment by alswl** [comment]
I use FitText.js resolved this problem.

```
    <script src="https://remarkjs.com/downloads/remark-latest.min.js"></script>
    <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.7.2/jquery.min.js"></script>
    <script src="https://cdn.staticfile.org/FitText.js/1.2.0/jquery.fittext.min.js"></script>

    <script>
        var slideshow = remark.create();
        $('.remark-slide-content').fitText(1.2, { minFontSize: '14px', maxFontSize: '36px' });
    </script>
```

quick command here:
https://github.com/alswl/.oOo./blob/master/local/bin/remark

**Comment by abelards** [comment]
Documented here, thanks all! https://github.com/gnab/remark/wiki/HTML-CSS-JS-samples
